doc/README.alluxio.md

```diff
@@ -0,0 +1,178 @@
+Instructions For Alluxio
+------------------------
+
+0) If necessary, download your favorite version of Alluxio and install
+   it into a location where it's accessible on all cluster nodes.
+   Usually this is on a NFS home directory.
+
+   See below in 'Alluxio Patching' about patches that are necessary for
+   correct Alluxio behavior. It is required when using Alluxio /bin scripts
+   with configuration directory other than the default ${ALLUXIO_HOME}/conf
+   (as Magpie does). This can be changed locally using ALLUXIO_CONF_DIR,
+   but it would affect only the master where the start command is running,
+   each worker looks for the configuration in the default directory. Applying
+   patches allows you to set the configuration file to the entire cluster.
+
+   See 'Convenience Scripts' in README about
+   misc/magpie-download-and-setup.sh, which may make the
+   downloading and patching easier.
+
+1) Select an appropriate submission script for running your job. You
+   can find them in the directory submission-scripts/, ex. Slurm
+   Sbatch scripts using srun are in script-sbatch-srun. To use Alluxio
+   in Magpie, it would be file magpie.sbatch-srun-alluxio or to use
+   Spark with Alluxio file magpie.sbatch-srun-spark-with-alluxio.
+
+   As Alluxio is predominantly a data caching service used by other
+   services (e.g. Spark), it's likely in the main submission scripts
+   for some of those big data projects. If the Alluxio section isn't
+   in it, just copy the Alluxio section from base script
+   (e.g. magpie.sbatch-srun) into it.
+
+2) Setup your job essentials at the top of the submission script. See
+   other projects (e.g. Spark in README.spark) for details on this
+   setup.
+
+3) Select how your job will run by setting MAGPIE_JOB_TYPE and/or
+   ALLUXIO_JOB. Initially, you'll likely want to set MAGPIE_JOB_TYPE to
+   'alluxio' and setting ALLUXIO_JOB to 'testalluxio'. This will allow
+   you to run a pre-written job to make sure things are setup correctly.
+
+4) Setup the essentials for Alluxio.
+
+   ALLUXIO_SETUP : Set to yes
+
+   ALLUXIO_VERSION : Set appropriately.
+
+   ALLUXIO_HOME : Where your Alluxio code is. Typically in an NFS
+   mount.
+
+   ALLUXIO_WORKER_ON_MASTER : Most big data projects do not have
+   a "slave" or "worker" that also runs on the master node
```

Comment:
per my comment about a possible Magpie 3.0 release timing for Alluxio, lets remove "slave" and just call it "worker" only.